lib/components/map/intermediate-places-overlay.js

```diff
@@ -0,0 +1,51 @@
+import { Marker } from 'react-leaflet'
+import React, {Component} from 'react'
+import { connect } from 'react-redux'
+
+import {
+  clearLocation,
+  forgetPlace,
+  rememberPlace,
+  setLocation
+} from '../../actions/map'
+import { getShowUserSettings } from '../../util/state'
+
+class IntermediatePlacesOverlay extends Component {
+  render () {
+    const {intermediatePlaces} = this.props
+    return (
+      <>
+        {Array.isArray(intermediatePlaces) && intermediatePlaces.map((place, i) => {
+          return place && place.lat && place.lon
+            ? <Marker key={i} position={[place.lat, place.lon]} />
```

Comment:
Whatever icon that is used for this marker should match up to the marker in the location-field and also the one shown in the itinerary result.